File: Code/Chapter 7/VB.NET/Exercise 7.4/Form1.vb
Imports System.Data
Imports System.Data.SqlClient

Public Class Form1
    Private myData As New DataSet
   Private Sub buttonSchema_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonSchema.Click
      Dim connectionString As String = _
       "Data Source=(local);Initial Catalog=Test;Integrated Security=SSPI;"

      Using testConnection As SqlConnection = New SqlConnection(connectionString)
         Dim testCommand As SqlCommand = testConnection.CreateCommand()
         testCommand.CommandText = "Select * from userTable; Select * from PermissionsTable"
         Dim dataAdapter As New SqlDataAdapter(testCommand)

         dataAdapter.FillSchema(myData, SchemaType.Source)
      End Using
      ' testConnection.Dispose called automatically.
      DisplayContents()
   End Sub
   Private Sub buttonData_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonData.Click
      Dim connectionString As String = _
       "Data Source=(local);Initial Catalog=Test;Integrated Security=SSPI;"

      Using testConnection As SqlConnection = New SqlConnection(connectionString)
         Dim testCommand As SqlCommand = testConnection.CreateCommand()
         testCommand.CommandText = "Select * from userTable; Select * from PermissionsTable"
         Dim dataAdapter As New SqlDataAdapter(testCommand)

         dataAdapter.Fill(myData)
      End Using
      ' testConnection.Dispose called automatically.
      DisplayContents()
   End Sub
    Private Sub DisplayContents()
        myData.WriteXml(Application.StartupPath & "\myData.Xml", XmlWriteMode.WriteSchema)
        xmlBrowser.Navigate(Application.StartupPath & "\myData.Xml")
    End Sub
End Class

File: Code/Chapter 7/C#/Exercise 7.4/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using System.Data.SqlClient;

#endregion

namespace Exercise_7_4
{
   partial class Form1 : Form
   {
      private DataSet myData;
      public Form1()
      {
         InitializeComponent();
         myData = new DataSet();
      }

      private void buttonSchema_Click(object sender, EventArgs e)
      {
         string connectionString =
         "Data Source=(local);Initial Catalog=Test;Integrated Security=SSPI;";

         using (SqlConnection testConnection = new SqlConnection(connectionString))
         {
            SqlCommand testCommand = testConnection.CreateCommand();
            testCommand.CommandText = "Select * from userTable; Select * from permissionsTable";
            SqlDataAdapter dataAdapter = new SqlDataAdapter(testCommand);

            dataAdapter.FillSchema(myData, SchemaType.Source);
         } // testConnection.Dispose called automatically.
         DisplayContents() ;
      }

      private void buttonData_Click(object sender, EventArgs e)
      {
         string connectionString =
         "Data Source=(local);Initial Catalog=Test;Integrated Security=SSPI;";

         using (SqlConnection testConnection = new SqlConnection(connectionString))
         {
            SqlCommand testCommand = testConnection.CreateCommand();
            testCommand.CommandText = "Select * from userTable; Select * from permissionsTable";
            SqlDataAdapter dataAdapter = new SqlDataAdapter(testCommand);

            dataAdapter.Fill(myData);
         } // testConnection.Dispose called automatically.
         DisplayContents() ;
      }

      private void DisplayContents()
      {
         myData.WriteXml(Application.StartupPath + "\\myData.Xml", XmlWriteMode.WriteSchema) ;
         xmlBrowser.Navigate(Application.StartupPath + "\\myData.Xml") ;
      }
   }
}